misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx

```diff
@@ -0,0 +1,237 @@
+import {
+  Card,
+  H3,
+  H5,
+  Icon,
+  Intent,
+  Menu,
+  MenuItem,
+  Tag,
+  Tooltip
+} from "@blueprintjs/core"
+import { IconNames } from "@blueprintjs/icons"
+import { FlexContainer } from "@misk/core"
+import { simpleSelect, simpleType } from "@misk/simpleredux"
+import * as React from "react"
+import { connect } from "react-redux"
+import styled from "styled-components"
+import { IDispatchProps, IState, rootDispatcher, rootSelectors } from "../ducks"
+
+export interface IWebAction {
+  applicationInterceptors: string[]
+  dispatchMechanism: string
+  function: string
+  functionAnnotations: string[]
+  name: string
+  networkInterceptors: string[]
+  parameterTypes: string[]
+  pathPattern: string
+  requestMediaTypes: string[]
+  responseMediaType: string
+  returnType: string
+}
+
+const HttpMethodTag = styled.span`
+  float: right;
+`
+
+const Column = styled.div`
+  flex-grow: 1;
+  flex-basis: 0;
+  min-width: 320px;
+`
+
+const MenuItemProps = {
+  multiline: true
+}
+
+const ComingSoon = () => (
+  <span>
+    <span>{"Coming soon "}</span>
+    <Icon icon={IconNames.AIRPLANE} />
+  </span>
+)
+
+const RequestResponeContentTypes = (props: { action: IWebAction }) => (
+  <span>
+    <span>{props.action.requestMediaTypes}</span>{" "}
+    <Icon icon={IconNames.ARROW_RIGHT} />{" "}
+    <span>{props.action.responseMediaType}</span>
+  </span>
+)
+
+enum HttpMethod {
+  "DELTE" = "DELETE",
+  "GET" = "GET",
+  "HEAD" = "HEAD",
+  "POST" = "POST",
+  "PATCH" = "PATCH",
+  "PUT" = "PUT"
+}
+
+const HttpMethodIntent = {
+  DELETE: Intent.DANGER,
+  GET: Intent.PRIMARY,
+  HEAD: Intent.WARNING,
+  PATCH: Intent.SUCCESS,
+  POST: Intent.SUCCESS,
+  PUT: Intent.SUCCESS
+}
+
+const filterAnnotations = (functionAnnotations: string[]) =>
+  functionAnnotations.filter(
+    a =>
+      !(
+        a.includes("RequestContentType") ||
```

Comment:
what's the significance of these?

Reply:
It filters out annotations that contain information that will already be displayed with dedicated panels (ie. don't need to show the AccessAnnotation because there's an `Access` box).

Reply:
ah gotcha. I think we'd probably want to filter this out server side if we think clients will find this not useful compared to other information provided by the API. Not a blocker for now!

Reply:
Sounds good! Definitely should upstream to the WebAction any filters / manipulations once we figure out what we'll need in the final UI 👍